src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandlerTest.java

```diff
@@ -710,4 +710,62 @@ private void assertValues(PreSubmitCallbackResponse<SscsCaseData> response, Inte
         assertThat(response.getData().getDateSentToDwp(), is(LocalDate.now().toString()));
         assertThat(response.getData().getDwpDueDate(), is(LocalDate.now().plusDays(expectedResponseDays).toString()));
     }
+
+    @Test
+    public void givenNoUploadedAndGeneratedDoc_thenReturnError() {
+        sscsCaseData.getDocumentGeneration().setGenerateNotice(NO);
+
+        final PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertFalse(response.getErrors().isEmpty());
+    }
+
+    @Test
+    public void givenGenerateNoticeIsYes_thenReturnCaseDataPreviewDoc() {
+        DocumentLink url = sscsCaseData.getDocumentStaging().getPreviewDocument();
+
+        handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        verify(footerService).createFooterAndAddDocToCase(eq(url), any(), any(), any(), any(), any(), any());
```

Comment:
can we check a specific  preview doc is in this verification?

Reply:
It might not work, because if we use the response from the AboutToSubmitHandler, preview doc will be set to null. So we won't be able to compare the preview doc that is set in case data, and the preview doc from the response